pkg/costmodel/costmodel.go

```diff
@@ -582,8 +580,19 @@ func (cm *CostModel) ComputeCostData(cli prometheusClient.Client, cp costAnalyze
 					ClusterID:       clusterID,
 					ClusterName:     cm.ClusterMap.NameFor(clusterID),
 				}
-				costs.CPUAllocation = getContainerAllocation(costs.CPUReq, costs.CPUUsed, "CPU")
-				costs.RAMAllocation = getContainerAllocation(costs.RAMReq, costs.RAMUsed, "RAM")
+
+				if len(costs.CPUReq) > 0 && len(costs.CPUUsed) > 0 {
+					costs.CPUAllocation = getContainerAllocation(costs.CPUReq[0], costs.CPUUsed[0], "CPU")
+				} else {
+					log.Warnf("CPU Requests or Usage data missing for %s", containerName)
```

Comment:
I would set this to `Debugf` probably, unless we have a strong feeling about this log level.

Reply:
Also, I'm not quite sure that this is the right logic. Let's think through it... we probably want to `getContainerAllocation` if either one of those two is present, right? We might want to even keep the old way, whereby we _always_ call it.

So I'm wondering if we want something more like:
```go
var cpuReq, cpuUse *util.Vector
if len(costs.CPUReq) > 0 {
    cpuReq = costs.CPUReq[0]
}
if len(costs.CPUUsed) > 0 {
    cpuUse = costs.CPUUsed[0]
}
costs.CPUAllocation(cpuReq, cpuUse, "CPU")
```
... and likewise for RAM. That maintains the old patter, whereby `getContainerAllocation` is always called, no matter what. And it lets `getContainerAllocation` handle `nil` cases, which it does quite aptly.

What do you think of that suggestion @thomasvn ?

Reply:
@nikovacevic I'm in agreement with both suggestions, thanks! I'll make the changes now.

Reply:
Just tested and pushed the changes!